docs/migration-service.md

```diff
@@ -122,96 +201,243 @@ flowchart TD
+5. **Blockchain Monitor**: Maintains current view of MPC contract state
+   - Embedded NEAR light client or RPC connection to track contract state
+   - Automatically detects events, e.g., migration initiations
+   - Enables autonomous operation without operator intervention
+
+6. **HTTP Server** (Optional): Operational monitoring and observability
+   - Health checks for liveness/readiness probes
+   - Prometheus-style metrics (keyshare freshness, backup success/failure rates)
+   - Operator dashboards for status visibility
+
+### Node Operator
+
+Node operators are responsible for:
+1. **Registering Backup Service**: Call `register_backup_service()` to store the backup service's public key in the contract
+2. **Initiating Migration**: Call `start_node_migration()` with the new node's `ParticipantInfo` when migrating to new hardware
+3. **Running Backup Service** (Soft Launch): Execute `backup-cli` scripts to backup and restore keyshares during migrations
+4. **Managing Environment Variables** (Soft Launch): Ensure `MPC_BACKUP_ENCRYPTION_KEY_HEX` is consistently set on both MPC node and backup-cli
+
+> **Hard Launch**: In hard launch, the backup service runs autonomously in a TEE and requires no manual intervention from operators beyond initial registration.
 
-_Note: The curious reader might ask why this protocol does not simply use the NEAR public/private key pairs associated to the MPC node and the backup service. The reason for this is is twofold:_
-- _those keys are meant for signature generation._
-- _While it is true that Curve25519 used in X25519 and edwards25519 used by the NEAR blockchain are [birationally equivalent](https://crypto.stackexchange.com/questions/43013/what-does-birational-equivalence-mean-in-a-cryptographic-context), so one could theoretically convert the NEAR account keys and use them for `X25519`, it is generally advised to use one key per application. This also allows us to use ephemeral keys, as opposed to static keys for the encryption. Which is desirable._
 
 
-#### Todo
-c.f. [(#949)](https://github.com/near/mpc/issues/949)
-- it is advised that the node operator grants access only to specific contract methods for the backup service and the node: [(#946)](https://github.com/near/mpc/issues/946)
-- define attestation for backup service [(#947)](https://github.com/near/mpc/issues/947)
+### Todo
+See [(#949)](https://github.com/near/mpc/issues/949)
+- It is advised that the node operator grants access only to specific contract methods for the backup service and the node: [(#946)](https://github.com/near/mpc/issues/946)
+
+**Hard Launch Implementation Tasks:**
+- [ ] Add `attestation` parameter to `register_backup_service()` contract method
+- [ ] Implement attestation verification for backup services in contract
+- [ ] Store attestation with expiration tracking in contract
+- [ ] Add attestation validity check in MPC node before backup/recovery operations
+- [ ] Develop backup service as long-running TEE application
```

Comment:
This is a big one :sweat_smile: we don't need to break it down here, but I'm thinking we could separate some steps. Analogous to the MPC nodes we can develop it as a standalone long-running application with mocked attestations first. This way we could get the blockchain interface, and all smart contract interactions working well in a controlled environment, and finally it would be easy to migrate it into a TEE when the time comes for that.

Reply:
I like the idea! I split it up a bit here: https://github.com/near/mpc/pull/1467/commits/174e3aae85c383e71923dabb6909fdd5b96ac129, although we’ll probably still refine it a bit before creating concrete GitHub issues from it.